nncf/quantization/algorithms/bias_correction/algorithm.py

```diff
@@ -401,27 +411,34 @@ def _remove_unnecessary_stats(self, position: int, subgraphs_data: Dict[str, Dic
         # Collects list of the statistics that needed for the future layers.
         needed_stats_list = []
         for i in range(position + 1, len(subgraphs_data)):
-            needed_stats_list.extend(subgraphs_data[i]["input_node_names"])
+            input_names = subgraphs_data[i]["subgraph_input_names"]
+            needed_stats_list.extend([self._collected_stat_inputs_map[name][0] for name in input_names])
 
-        node_inputs_name = subgraphs_data[position]["input_node_names"]
+        node_inputs_name = subgraphs_data[position]["subgraph_input_names"]
         for node_input_name in node_inputs_name:
-            if node_input_name not in needed_stats_list and node_input_name in self._fp_inputs:
-                nncf_logger.debug(f"Dropped {node_input_name}")
-                self._fp_inputs[node_input_name] = []
+            activation_name, _ = self._collected_stat_inputs_map[node_input_name]
+            if activation_name not in needed_stats_list and activation_name in self._fp_inputs:
+                nncf_logger.debug(f"Dropped {activation_name} output statistics.")
+                self._fp_inputs[activation_name] = []
 
-    def _get_fp_inputs(self, statistic_points: StatisticPointsContainer, node_name: str) -> np.ndarray:
+    def _get_fp_inputs(self, statistic_points: StatisticPointsContainer, node_name: str, port_id: int) -> np.ndarray:
         """
         Makes out pre-layer needed data from the floating-point collected statistics.
 
         :param statistic_points: Filled StatisticPointsContainer.
         :param node_name: Name of the current layer.
+        :param port_id: Port id for statistics collection.
         :return: Collected mean tensor data and shape for the further bias calculation.
         """
 
         def input_filter_func(point):
+            # For the floating-point statistics collected in POST_LAYER style,
+            # we also need to determine the output port id.
+            # For the cases when the layer has more than one (0) output port.
```

Comment:
Should be added same filter for FBC?

Reply:
It was already added. Check the FBC code.